### DISCUSSION TITLE
Assistance with the new Functions Component

### DISCUSSION THREAD

**Discussion by edleeman17** [digitalocean/terraform-provider-digitalocean#836] [Q&A] [answered]
Hi All,

Apologies if this is something trivial. I have been playing with the new functions component in the 2.20.0 version of the provider.

I am using the following TF code to create a brand new function.

```terraform
resource "digitalocean_app" "stripe-listener-function" {
  spec {
    name   = "stripe-listener"
    region = "lon"
    domain {
      name = "backend.example.com"
    }

    function {
      name = "stripe-listener"
      git {
        repo_clone_url = "https://github.com/digitalocean/sample-functions-nodejs-helloworld.git"
        branch = "master"
      }
      routes {
        path = "/order-processor"
      }
    }
  }
}
```

I would expect this to show up under https://cloud.digitalocean.com/functions/ but instead it is showing as an App...

Expectation:
![image](https://user-images.githubusercontent.com/61695425/173080304-eaefae8a-a11a-40ce-b772-566886a98812.png)

Actual:
![image](https://user-images.githubusercontent.com/61695425/173080684-1f1f65dc-f7b6-4391-bc6c-1b9d6030fbce.png)

Is this expected? Am I doing something wrong?

Many thanks,
Ed

**Comment by andrewsomething** [comment, marked as answer]
Hi Ed,

This is the expected behavior. Currently, Terraform only supports functions as components of App Platform apps. Apps can contain multiple components, allowing you to compose an application from a number of different pieces like static sites, functions, and containerized services.

Currently, if you want to create standalone functions outside of App Platform, you can use the control panel or the `doctl` CLI.

**Reply by edleeman17** [reply]
Amazing, thank you for your quick response
